cobc/parser.y

```diff
@@ -314,24 +314,20 @@ static COB_INLINE void
 check_area_a (cb_tree stmt) {
 	if (!cobc_in_area_a && cobc_areacheck) {
 		if (stmt)
-			cb_warning_x (COBC_WARN_FILLER, stmt,
-				      _("'%s' should start in Area A"),
-				      CB_NAME (stmt));
+			(void) cb_syntax_check_x (stmt, _("'%s' should start in Area A"),
+						  CB_NAME (stmt));
 		else
-			cb_warning (COBC_WARN_FILLER,
-				    _("statement should start in Area A"));
+			(void) cb_syntax_check (_("statement should start in Area A"));
```

Comment:
I've wondered before: is this really a `stmt`->`statement` or something else (paragraph/procedure?)

Reply:
Hum indeed… in general that can either be a word (section/paragraph name), or NULL (in which case it's typically a statement or a data division entry). So we may get "statement should start in Area A" for data items: for those we may duplicate `check_area_a` and use a specialized message.